props_lab/src/components/donationForm.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import RecentDonations from './recentDonations';
+
+class DonationForm extends React.Component {
+    state = {
+        name: "",
+        amount: 0,
+        caption: "",
+        recentDonators: []
+    }
+
+
+    handleNameChange = (e) => {
```

Comment:
You can replace all these handle changes by naming all your components the same name as the state youre trying to update. Then, you can simply have one handleChange, like this...


`this.setState({[e.target.name]: e.target.value})`